src/components/SearchableSelect/SearchableSelect.jsx

```diff
@@ -0,0 +1,427 @@
+import React from 'react';
+import _ from 'lodash';
+import { createClass, findTypes } from '../../util/component-types';
+import { lucidClassNames } from '../../util/style-helpers';
+import { partitionText } from '../../util/text-manipulation';
+import * as reducers from './SearchableSelect.reducers';
+import CaretIcon from '../Icon/CaretIcon/CaretIcon';
+import DropMenu from '../DropMenu/DropMenu';
+import LoadingIcon from '../Icon/LoadingIcon/LoadingIcon';
+import SearchField from '../SearchField/SearchField';
+
+const cx = lucidClassNames.bind('&-SearchableSelect');
+
+const {
+	any,
+	bool,
+	func,
+	node,
+	number,
+	object,
+	shape,
+	string,
+	oneOfType,
+} = React.PropTypes;
+
+/**
+ *
+ * {"categories": ["controls", "selectors"], "madeFrom": ["DropMenu"]}
+ *
+ * A selector control (like native `<select>`) which is used to select a single option from a dropdown list.
+ * Supports option groups with and without labels.
+ */
+const SearchableSelect = createClass({
```

Comment:
I vote for `SingleSelectFilterable`. There were some questions about async search using this component, but I don't think that makes sense given that it doesn't do paging or infinite scroll or anything. I think this component is really only appropriate for cases where you can show the entire list at once, i.e. places where you could use an regular html `select`.